tsdb/head.go

```diff
@@ -1923,33 +1944,40 @@ func (s *stripeSeries) gc(mint int64, minOOOMmapRef chunks.ChunkDiskMapperRef) (
 		deletedForCallback[series.ref] = series.lset
 	}
 
-	// Run through all series shard by shard, checking which should be deleted.
+	s.iter(func(i int, hash uint64, series *memSeries, deletedForCallback map[chunks.HeadSeriesRef]labels.Labels) {
+		check(i, hash, series, deletedForCallback)
+	}, func(deletedForCallback map[chunks.HeadSeriesRef]labels.Labels) {
+		s.seriesLifecycleCallback.PostDeletion(deletedForCallback)
+	})
+
+	if actualMint == math.MaxInt64 {
+		actualMint = mint
+	}
+
+	return deleted, rmChunks, actualMint, minOOOTime, minMmapFile
+}
+
+func (s *stripeSeries) iter(f func(int, uint64, *memSeries, map[chunks.HeadSeriesRef]labels.Labels), endShard func(map[chunks.HeadSeriesRef]labels.Labels)) {
```

Comment:
Let's not have "deletion" in the name, because this function in itself has nothing to do with deletion. The users of this just happen to use it for deletion.

Reply:
If we dont have deletion on the name we cannot call the post deletion hook inside the function! =/ What u guys think is better here?